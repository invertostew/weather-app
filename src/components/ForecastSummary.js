import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";
import moment from "moment";

import "../styles/ForecastSummary.css";

function ForecastSummary({
  date,
  icon,
  temperature,
  description,
  handleForecastSelect,
}) {
  return (
    <article className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary__date">
        <p className="forecast-summary__content forecast-summary__content--date">
          {moment(date).format("ddd Do MMM")}
        </p>
      </div>

      <div className="forecast-summary__icon" data-testid="forecast-icon">
        <WeatherIcon
          name="owm"
          iconId={icon}
          className="forecast-summary__content forecast-summary__content--icon"
        />
      </div>

      <div className="forecast-summary__temperature">
        <p className="forecast-summary__content forecast-summary__content--temperature">
          {temperature.max} &deg;C
        </p>
      </div>

      <div className="forecast-summary__description">
        <p className="forecast-summary__content forecast-summary__content--description">
          {description}
        </p>
      </div>

      <div className="forecast-summary__more-details">
        <button
          type="button"
          className="forecast-summary__button"
          data-date={date}
          onClick={handleForecastSelect}
        >
          More Details
        </button>
      </div>
    </article>
  );
}

ForecastSummary.propTypes = {
  date: PropTypes.number.isRequired,
  icon: PropTypes.string.isRequired,
  temperature: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number.isRequired,
  }).isRequired,
  description: PropTypes.string.isRequired,
  handleForecastSelect: PropTypes.func.isRequired,
};

export default ForecastSummary;
