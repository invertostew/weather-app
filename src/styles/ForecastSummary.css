.forecast-summary {
  flex-grow: 1;
  text-align: center;
}

.forecast-summary__content {
  margin: 0;
}

.forecast-summary__content--icon {
  font-size: 3rem;
  margin: 2rem 0;
}

.forecast-summary__content--temperature {
  margin-bottom: 0.5rem;
}

.forecast-summary__button {
  border: 2px solid #153253;
  border-radius: 99px;
  padding: 0.7rem 1.5rem;
  font-weight: 700;
  color: #fff;
  background-color: #153253;
  cursor: pointer;
  margin-top: 1rem;
}
