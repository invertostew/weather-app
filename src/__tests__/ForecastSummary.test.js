import { render } from "@testing-library/react";

import ForecastSummary from "../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1111111,
    icon: "800",
    temperature: {
      min: 10,
      max: 20,
    },
    description: "Test Description",
    handleForecastSelect: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        icon={validProps.icon}
        temperature={validProps.temperature}
        description={validProps.description}
        handleForecastSelect={validProps.handleForecastSelect}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary
        date={validProps.date}
        icon={validProps.icon}
        temperature={validProps.temperature}
        description={validProps.description}
        handleForecastSelect={validProps.handleForecastSelect}
      />
    );

    expect(getByText("Thu 1st Jan")).toHaveClass(
      "forecast-summary__content--date"
    );
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary__icon");
    expect(getByText("20 Â°C")).toHaveClass(
      "forecast-summary__content--temperature"
    );
    expect(getByText("Test Description")).toHaveClass(
      "forecast-summary__content--description"
    );
    expect(getByText("More Details")).toHaveClass("forecast-summary__button");
  });
});
