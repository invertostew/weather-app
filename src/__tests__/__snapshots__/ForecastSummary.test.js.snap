// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <article
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      <p
        class="forecast-summary__content forecast-summary__content--date"
      >
        Thu 1st Jan
      </p>
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        class="wi wi-day-sunny forecast-summary__content forecast-summary__content--icon"
        name="owm"
      />
    </div>
    <div
      class="forecast-summary__temperature"
    >
      <p
        class="forecast-summary__content forecast-summary__content--temperature"
      >
        20 Â°C
      </p>
    </div>
    <div
      class="forecast-summary__description"
    >
      <p
        class="forecast-summary__content forecast-summary__content--description"
      >
        Test Description
      </p>
    </div>
    <div
      class="forecast-summary__more-details"
    >
      <button
        class="forecast-summary__button"
        data-date="1111111"
        type="button"
      >
        More Details
      </button>
    </div>
  </article>
</DocumentFragment>
`;
